File: src/Java/Creational Patterns/Factory/CreditCards/src/Creators/TitaniumCreditCardCreator.java
package Creators;

import Products.CreditCard;
import Products.Titanium;

import java.math.BigDecimal;

public class TitaniumCreditCardCreator implements ICreditCardCreator {
    @Override
    public CreditCard createCreditCard(String model, BigDecimal limit, BigDecimal annualCharge) {
        return new Titanium(model, limit, annualCharge);
    }
}

File: src/JavaScript/CreationalPatterns/Factory/CreditCards/creators/titaniumCreditCardCreator.js

const ICreditCardCreator = require('./creditCardCreator');
const Titanium = require('../products/titanium');

/**
 * @implements {ICreditCardCreator}
 */
class TitaniumCreditCardCreator {
    /**
     * @inheritDoc
     */
    createCreditCard(model, limit, annualCharge) {
        return new Titanium(model, limit, annualCharge);
    }
}

module.exports = TitaniumCreditCardCreator;